pkg/products/threescale/reconciler.go

```diff
@@ -434,6 +433,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, installation *integreatlyv1a
 		return phase, err
 	}
 
+	// Ensure ratelimit annotation is ready before returning phase complete
+	phase, err = r.reconcileRatelimitPortAnnotation(ctx, serverClient)
```

Comment:
hey @bryankeane0 any reason to have it at the end of reconcile loop instead of BEFORE creating/reconciling rate limiting?
Reason I say that is because it might turn out that the rate limiting will be created, dc's update, then update again because of 3scale operator rollback, then we are switching off the reconciliation and only then update the dcs again from rate limiting?

Reply:
Cool I'll implement it now thanks for the suggestion